crates/telio-traversal/src/session_keeper.rs

```diff
@@ -219,14 +235,23 @@ impl Runtime for State {
             Ok((pk, action)) = self.keepalive_actions.select_action() => {
                 let pk = *pk;
 
-                telio_log_trace!("({}) name: \"{}\", action = actions.select_action()", Self::NAME, pk);
+                if let Some(last_handshake_time_provider) = &self.last_handshake_time_provider {
```

Comment:
How come we use the old logic in case `last_handshake_time_provider` is `None`? Just trying to understand

Reply:
The `last_handshake_time_provider` is set in the device/mod.rs` based on the feature config flag `skip_unresponsive_peers`. If it is set no `None` we should not skip such peers and always send ping, otherwise we should check respective last handshake time and decide based on that if we should send the ping.

Reply:
Your are right, ping sending logic should have been duplicated in the outer else branch.

Reply:
Fixed the code and expanded the relevant test to cover both old and new way of working.